hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -157,7 +160,23 @@ class Handle
     // END
   }
 
-  [[nodiscard]] bool get_value(double & value) const
+  template <typename T = double>
+  [[nodiscard]] std::optional<T> get_value() const
+  {
+    std::shared_lock<std::shared_mutex> lock(handle_mutex_, std::try_to_lock);
+    if (!lock.owns_lock())
+    {
+      return std::nullopt;
+    }
+    THROW_ON_NULLPTR(this->value_ptr_);
+    // TODO(saikishor): uncomment the following line when HANDLE_DATATYPE is used with multiple
+    // datatypes and remove the line below it.
+    // return value_ptr_ != nullptr ? *value_ptr_ : std::get<T>(value_);
+    return *value_ptr_;
+  }
+
+  template <typename T>
+  [[nodiscard]] bool get_value(T & value) const
```

Comment:
Do we actually need this format of `get_value`? Are we using this somewhere of do you have specific use-case where this has advantages over `std::optional<T> get_value()`?

Reply:
Right now, we are not using it in ros2_control repositories, but this exists from many releases both in rolling nd jazzy. So, at least not to break API of people that might use it, I haven't removed it here

Reply:
We should then deprecate this.

Reply:
I'm not sure if we should deprecate this. It was added long back to have the similar convention as the `set_value` method. If you still think, this should be deprecated, then we can deprecate it

Reply:
to the WG meeting, bring your dueling pistols!